recipes/wasmtime/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class WasmtimeConan(ConanFile):
+    name = 'wasmtime'
+    homepage = 'https://github.com/bytecodealliance/wasmtime'
+    license = 'Apache License 2.0'
+    url = 'https://github.com/conan-io/conan-center-index'
+    description = "Standalone JIT-style runtime for WebAssembly, using Cranelift"
+    topics = ("webassembly", "wasm", "wasi")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        'fPIC': [True],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+    }
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build(self):
+        try:
+            if self.settings.arch == "armv8" and self.settings.os == "Android":
+                os_name = "Linux"
+            else:
+                os_name = str(self.settings.os)
+
+            archive_ext = "zip" if os_name == "Windows" else "tar.xz"
+            url = f"https://github.com/bytecodealliance/wasmtime/releases/download/v{self.version}/wasmtime-v{self.version}-{self.settings.arch}-{os_name.lower()}-c-api.{archive_ext}"
+            tools.get(url, strip_root=True, destination=self._source_subfolder)
+        except:
+            raise Exception("Binary does not exist for these settings")
```

Comment:
This should be moved to the validate method... although it's limited

Reply:
According the conan documentation https://docs.conan.io/en/latest/creating_packages/existing_binaries.html I've added it in right place

Reply:
CCI is a very specialized build system and our best practices are different from the generic advice in the docs. CCI prefers using validate to improve it's workflow

Reply:
Done